corehq/apps/app_manager/tasks.py

```diff
@@ -57,7 +58,9 @@ def prune_auto_generated_builds(domain, app_id):
              .source(['_id']))
 
     for hit in query.run().hits:
-        if hit['_id'] == last_build:
+        if hit['_id'] == last_build_id:
             continue
         app = get_app(domain, hit['_id'])
+        if not app.is_auto_generated or app.copy_of != app_id or app.id == last_build_id:
```

Comment:
should `app.id` read `app._id`?

Reply:
They seem to be the same...that's something I've never understood, when to use `thing._id` and when to just use `thing.id`

Reply:
interesting, it looks like that works because [`VersionedDoc`](https://github.com/dimagi/commcare-hq/blob/88ccaa70d31dad97dd966348cc6c3d6e49fd89c2/corehq/apps/app_manager/models.py#L4549-L4551) makes it possible. otherwise normally for a subclass of `Document` it would be only `_id`. fine to leave as is

Reply:
cool, thanks for investigating